aten/src/ATen/cpu/vml.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "ATen/Config.h"
+#include "ATen/Parallel.h"
+#include "ATen/cpu/vec256/functional.h"
+#include "ATen/cpu/vec256/vec256.h"
+
+// This header implements various unary operations using a MKL VML style
+// interface.
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+
+#if AT_MKL_ENABLED()
+#include <mkl.h>
+#endif
+
+namespace at {
+namespace vml {
+namespace {
+
+using namespace vec256;
+
+template <typename scalar_t>
+inline void vabs(scalar_t* out, scalar_t* in, int64_t size) {
+  parallel_for(0, size, 2048, [out, in](int64_t begin, int64_t end) {
+    map([](const Vec256<scalar_t>& x) { return x.abs(); },
+        out + begin,
+        in + begin,
+        end - begin);
+  });
+}
```

Comment:
Can't this be implemented using `IMPLEMENT_VML`?

Reply:
Coming right up in another PR.